docs-ref-conceptual/get-started-tutorial-3-use-variables.md

```diff
@@ -0,0 +1,305 @@
+---
+title: Get variable values from Azure resources or a local JSON file | Microsoft Docs
+description: Learn how to use variables to store JSON file properties and query output.
+manager: jasongroce
+author: dbradish-microsoft
+ms.author: dbradish
+ms.date: 11/15/2023
+ms.topic: conceptual
+ms.service: azure-cli
+ms.tool: azure-cli
+ms.custom: devx-track-azurecli
+keywords: azure, 
+---
+# Populate variables for use in scripts
+
+In this tutorial step you will learn how to get information from a local JSON file, new and existing Azure resources, and store the information in a variable. The variable can then be used in Azure CLI to perform build and destroy jobs at scale. Some common use-cases are:
+
+* Get information about an existing Azure resource, such as a resource ID.
+* Get output from an Azure CLI command, such as a password.
+* Get JSON objects for environment information, such as development, stage and production IDs.
+
+## Prerequisites
+
+* You have access to a resource group and storage account with `contributor` permissions.
+
+## Get command output using JMESPath query
+
+There are times when you want to get information about an existing Azure resource and return that information to your console screen, or store it in a variable for use within a script. In the Azure CLI, use the `--query` parameter to execute a [JMESPath query](https://jmespath.org/) to perform these tasks.
+
+> [!TIP]
+> The syntax for `--query` is case sensitive _and environment-specific_.  If you receive empty results, check your capitalization. Avoid quoting errors by applying the rules you learned in [Write Azure CLI commands for different environments](./get-started-tutorial-2-environment-syntax.md)
+
+Unless the `--output` parameter is specified, these examples rely on a default output configuration of `json` set in [Prepare your environment for the Azure CLI](./get-started-tutorial-1-prepare-environment.md)
+
+### Get JSON dictionary properties of an Azure resource
+
+Using the storage account created in [Learn Azure CLI syntax differences in Bash, PowerShell and Cmd](./get-started-tutorial-2-environment-syntax.md), get the `primaryEndpoints` of your new storage account.
+
+```azurecli-interactive
```

Comment:
This will return: ERROR: (--name | --ids) are required. Perhaps this example should also include those fields